ST_mortality.c

```diff
@@ -68,70 +71,74 @@ void _kill_maxage(void);
 
 /************ File-Level Variable Declarations *************/
 /***********************************************************/
+/**
+ * \brief TRUE if a function in ST_mortality.c killed an individual.
+ * \ingroup MORTALITY_PRIVATE
+ */
 Bool _SomeKillage;
-/* flag: some plant was reduced and PR is affected. */
-/* 7/5/01  - currently flag is set but unused. */
+
 extern
   pcg32_random_t mortality_rng; //declared in ST_main.c
 
-
-/***********************************************************/
-/***********************************************************/
+/**
+ * \brief Performs mortality that occurs during the growing season.
+ * 
+ *  This function reduces "plant-space" according to
+ *  resource availability, age, growth rate, and disturbance
+ *  conditions.  Also, succulents are "mortified" if it's a wet
+ *  year.
+ * 
+ *  This function is not to be confused with mort_EndOfYear().
+ *
+ *  An important consideration is the fact that the age of new
+ *  plants starts at 1.  Consider that for it to establish, it
+ *  has already survived the massive mortality of year 0; this
+ *  rate of year 0 mortality, among other factors of survival,
+ *  is captured by the probability of establishment (see
+ *  rgroup_Establish()).  However, for consistency with other
+ *  arrays, the age-related arrays are indexed from 0.  Thus,
+ *  age is base1 but the arrays are base0.
+ * 
+ *  The outline for this function is:
+ *    1) Death by limited resources.
+ *    2) Age independent mortality.
+ *    3) Slow growth mortality.
+ *    4) Succulent wet season mortality.
+ *    5) Death by pats, mounds or burrows.
+ *
+ *  More specifically:
+ *  - Process each group separately.
+ *  - If resources required for a group are greater than available
+ *    (PR > 1) for the maximum years stretching is allowed, kill
+ *    plants according the rules for insufficient resources.  If
+ *    PR < 1, reset number of stretched years to zero.  This means
+ *    that even one year of adequate resources nulls any number of
+ *    years of stretched resources.
+ *  - For each species, if the user requested the mortality functions
+ *    to be executed, run the age-independent and slow-growth
+ *    routines.
+ *  - If the current species is a succulent and this is a wet year,
+ *    execute the mortality function for succulents.  That is,
+ *    reduce each plant by some constant proportion, defined by
+ *    eqn 16.  The parameters are defined in the group-parms and
+ *    the reduction amount is computed in the Env module based
+ *    on precipitation.
+ *  - execute disturbance effects, if any.
```

Comment:
- Execute disturbances from fecal pats, ant mounds, and animal burrows, if any.

Point here is that other disturbances (i.e. fire and grazing) are implemented elsewhere.

Reply:
Addressed by commit 984bd13.